db_sql.go

```diff
@@ -339,8 +339,14 @@ func (x *sqlUserStoreDB) RenameIdentity(oldIdent, newIdent string) error {
 	}
 }
 
-func (x *sqlUserStoreDB) GetIdentities() ([]AuthUser, error) {
-	rows, err := x.db.Query("SELECT userid, email, username, firstname, lastname, mobile, phone, remarks, created, createdby, modified, modifiedby, authusertype FROM authuserstore WHERE (archived = false OR archived IS NULL)")
+func (x *sqlUserStoreDB) GetIdentities(getIdentitiesFlag GetIdentitiesFlag) ([]AuthUser, error) {
+	var sqlStatement string
+	if getIdentitiesFlag == GetIdentitiesFlagDeleted {
```

Comment:
Use binary and to check for field presence.

Also, make the common part of the query a common string. And only add that final "WHERE" clause if the flag is not present.

Reply:
`if (getIdentitiesFlag && GetIdentitiesFlagDeleted) `

Reply:
Don't listen to him!
`if getIdentitiesFlag & GetIdentitiesFlagDeleted !=0`